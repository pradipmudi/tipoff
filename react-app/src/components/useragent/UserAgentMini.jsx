import React from 'react';

import BrowserIcon from './BrowserIcon';
import OsIcon from './OsIcon';
import DeviceIcon from './DeviceIcon';

const UserAgentMini = ({ua}) => (
    <span>
        <BrowserIcon browser={ua ? ua.browser : null} />
        <OsIcon os={ua ? ua.os : null} />
        <DeviceIcon device={ua ? ua.device : null} />
    </span>
);

export default UserAgentMini;