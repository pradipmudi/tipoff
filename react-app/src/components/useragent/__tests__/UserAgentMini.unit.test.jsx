import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import UserAgentMini from '../UserAgentMini';


describe("UserAgentMini", () => {
    const ua = {
        ua: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
        browser: { name: 'Chrome', version: '79.0.3945.88', major: '79' },
        engine: { name: 'Blink', version: '79.0.3945.88' },
        os: { name: 'Windows', version: '10' },
        device: { vendor: undefined, model: undefined, type: undefined },
        cpu: { architecture: 'amd64' }
    }

    const wrapper = shallow(<UserAgentMini ua={ua} />);

    it("matches snapshot", () => {
        const snap = renderer.create(<UserAgentMini ua={ua} />).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("renders without issues", () => {
        expect(wrapper.exists()).toEqual(true);
    });

    it("renders browser icon", () => {
        expect(wrapper.find('BrowserIcon').exists()).toEqual(true);
    });

    it("renders os icon", () => {
        expect(wrapper.find('OsIcon').exists()).toEqual(true);
    });

    it("renders device icon", () => {
        expect(wrapper.find('DeviceIcon').exists()).toEqual(true);
    });
});