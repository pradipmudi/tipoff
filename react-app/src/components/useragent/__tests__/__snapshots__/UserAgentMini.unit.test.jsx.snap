// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAgentMini matches snapshot 1`] = `
<span>
  <span>
    <span
      className="icon"
      title="Chrome:79"
    >
      <i
        aria-hidden="true"
        className="fab fa-chrome"
      />
    </span>
  </span>
  <span>
    <span
      className="icon"
      title="Windows:10"
    >
      <i
        aria-hidden="true"
        className="fab fa-windows"
      />
    </span>
  </span>
  <span>
    <span
      className="icon"
    >
      <i
        aria-hidden="true"
        className="fas fa-laptop"
      />
    </span>
  </span>
</span>
`;
